### PR TITLE
Added beam search & fixed open list clear bug.

### PR DESCRIPTION
**Pull request by snlemons**
Added an implementation for beam search algorithm and fixed an issue where open list's clear function wasn't resetting the "fill" variable.

### CODE REVIEW

File: search/beam.hpp [hunk 1]
```diff
@@ -0,0 +1,213 @@
+// Copyright © 2013 the Search Authors under the MIT license. See AUTHORS for the list of authors.
```
Comment (eaburns) [inline search/beam.hpp:1]: year 2020, not 2013

File: search/beam.hpp [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    closed.prstats(stdout, "closed ");
+    dfpair(stdout, "open list type", "%s", open.kind());
+    dfpair(stdout, "node size", "%u", sizeof(Node));
+  }
+
+
+private:
+
+	void expand(D &d, Node *n, State &state) {
+		SearchAlgorithm<D>::res.expd++;
```
Comment (eaburns) [inline search/beam.hpp:146]: the indents changed here from spaces before to tabs. i don't recall what the other files in the repo use, but prefer sticking to that

File: search/greedy.hpp [hunk 3]
```diff
@@ -52,6 +56,11 @@ template <class D, bool speedy = false> struct Greedy : public SearchAlgorithm<D
 	Greedy(int argc, const char *argv[]) :
 		SearchAlgorithm<D>(argc, argv), closed(30000001) {
 		nodes = new Pool<Node>();
+        dropdups = false;
```
Comment (eaburns) [inline search/greedy.hpp:59]: optional:

dropdups = strcmp(...) == 0;

**Comment by eaburns** [comment]
I mostly looked superficially; I don't really remember this code will, and unfortunately don't have much spare time to come back up to speed on it enough to do a thorough review. One thing I noticed is that indentation here is inconsistent between spaces and tabs. I don't recall which this repo used, but it'll be much easier to read if you can stick to that one.

Also, since I'm now useless, feel free to fork this repo and start your own. I'll also add you as a contributor if you want to just keep committing here.

**Comment by snlemons** [comment]
I'll check out the tab settings and make them consistent with existing code.

Looks like the contributor invitation expired, as I didn't get a notofication for it. Any chance you could re-send?

**Comment by eaburns** [comment]
Sure. I think I just re-sent it.

**Pull request closed without merging**